library/swift/EngineBuilder.swift

```diff
@@ -216,6 +217,17 @@ public class EngineBuilder: NSObject {
     return self
   }
 
+  /// Set a closure to be called when the engine emits an event.
+  ///
+  /// - parameter closure: The closure to be called.
+  ///
+  /// - returns: This builder.
+  @discardableResult
+  public func setEventTracker(closure: @escaping ([String: String]) -> Void) -> Self {
```

Comment:
@rebello95 @goaway @junr03 

I tried to come up with an integration test for `setEventTracker` that would look similar to [SetLoggerTest](https://github.com/envoyproxy/envoy-mobile/blob/main/test/swift/integration/SetLoggerTest.swift) test. 

There are some issues here since by default Envoy Engine does not emit any events - we will emit them in our private extensions to Envoy Mobile. That means that for the integration test to work we would need to "simulate" an emission of an event using `Api::External::retrieveApi("envoy_event_tracker_api_name")` that's written in C++ and not accessible in Swift (a language the integration test would be written in).

Any ideas for how I could test `setEventTracker` functionality that's being added in this PR?

Reply:
How about adding a native test filter that makes use of this and then add that via `addNativeFilter`? That way you can invoke the event tracker from there and verify that sending a request through EM triggers an event

Reply:
That could work - I assume I would need to perform some request for the filter chain to be executed. Does it sound about right or is there an easier way to do this?